doc: Improve NixOS documentation (#20497)
vectordotdev/vector: website/content/en/docs/setup/installation/operating-systems/nixos.md

Nixpkgs has a [community maintained module][nixpkg-vector] for Vector, the
options for which may be viewed on the [NixOS Search][nixos-search].

> This can be used to deploy and configure Vector on a NixOS system, the module
> will also verify that the Vector configuration is valid before enabling any
> changes.

Reviewer: ```suggestion
This can be used to deploy and configure Vector on a NixOS system. The module also verifies that the Vector configuration is valid before enabling any
changes.
```

For example, place into a system's `configuration.nix`:

```nix
services.vector = {
  enable = true;
  journaldAccess = true;
  settings = {
    sources = {
      journald.type = "journald";

      vector_metrics.type = "internal_metrics";
    };

    sinks = {
      loki = {
        type = "loki";
        inputs = [ "journald" ];
        endpoint = "https://loki.mycompany.com";

        labels.source = "journald";
      };

      prometheus_exporter = {
        type = "prometheus_exporter";
        inputs = [ "vector_metrics" ];
        address = "[::]:9598"
      };
    };
  };
};
```

> Occasionally, it'll be necessary to provide Vector with additional permissions
> to access files which belong to other services. Below is an example in which
> Vector is granted access to log files which belong to [Caddy](caddy):

Reviewer: ```suggestion
Occasionally, it is necessary to provide Vector with additional permissions
to access files that belong to other services. Below is an example in which
Vector is granted access to log files that belong to [Caddy](caddy):
```
